airflow/jobs.py

```diff
@@ -535,7 +535,10 @@ def process_dag(self, dag, queue):
             elif ti.is_runnable(flag_upstream_failed=True):
```

Comment:
From reading a bit of code just now, `is_runnable` should include a task that is `UP_FOR_RETRY` where the retry_delay has expired, in which case it wouldn't even hit the `elif ti.is_premature():`